### ISSUE TITLE
the Runnable test is bonkers

### ISSUE THREAD

**Issue by boneskull** [mochajs/mocha#3342]
in the `Runnable` suite (`test/unit/runnable.spec.js`) we are poisoning the global `setTimeout`, `setInterval`, etc., to assert that we don't call those functions ([read the comments](https://github.com/mochajs/mocha/blob/master/test/unit/runnable.spec.js) for further explanation of why this is necessary).

but, really, that problem is better solved with linting.  **write a one-off eslint plugin/rule that ensures Mocha never calls these functions internally, once the test files begin loading**.

In other words, this is allowed:

```js
var setTimeout = global.setTimeout;
setTimeout(something);
// or
var setTimeout = require('timers').setTimeout;
setTimeout(something);
```

This is **not** allowed:

```js
global.setTimeout(something);
// or
setTimeout(something);
// or
require('timers').setTimeout(something);
```

**Comment by plroebuck** [comment]
Should this issue already be closed?
